shop/models.py

```diff
@@ -0,0 +1,137 @@
+        # Восстановление "удаленной" строки
+        self.deleted_at = None
+        self.save()
+
+    @property
+    def is_deleted(self):
+        return self.deleted_at is not None
+
+
+#The model with categories of products
+class Category(BaseModel):
+    name = models.CharField(max_length=255, verbose_name='Имя категории')
+    description = models.TextField(blank=True, verbose_name='Описание')
+
+    class Meta:
+        verbose_name = "Категории"
+        verbose_name_plural = "Категории"
+        ordering = ['name']
+        indexes = [
+            models.Index(fields=['name'])
+        ]
+
+    def __str__(self):
+        return self.name
+
+
+#The model with subcategories of products
+class SubCategory(BaseModel):
+    category = models.ForeignKey(Category, related_name='category', on_delete=models.CASCADE)
+    name = models.CharField(max_length=255, verbose_name='Имя подкатегории')
+    description = models.TextField(blank=True, verbose_name='Описание')
+
+    class Meta:
+        verbose_name = "Подкатегории"
+        verbose_name_plural = "Подкатегории"
+
+    def __str__(self):
+        return f"{self.category} - {self.name}"
+
+
+#The model with products
+class Product(BaseModel):
+    name = models.CharField(max_length=255, verbose_name='Товар')
+    photo = models.ImageField(upload_to='product', default='Add you photo...', verbose_name='Фото')
+    description = models.TextField(verbose_name='Описание')
+    price = models.DecimalField(max_digits=10, decimal_places=2, verbose_name='Цена')
+    subcategory = models.ForeignKey(SubCategory, related_name='products', on_delete=models.CASCADE, verbose_name='Подкатегория')
+    shop = models.ForeignKey('Shop', related_name='shops', on_delete=models.CASCADE, verbose_name='Магазин')
+    stock = models.PositiveIntegerField(verbose_name='В наличии')
+
+    class Meta:
+        verbose_name = 'Товар'
+        verbose_name_plural = 'Товары'
+
+    def __str__(self):
+        return self.name
+
+
+#The model with salesmans
+class Shop(models.Model):
+    seller = models.OneToOneField(Seller, on_delete=models.CASCADE, related_name='shop')
+    shop_name = models.CharField(max_length=100, verbose_name='Продавец')
+    country = models.CharField(max_length=40, verbose_name='Страна')
+    category = models.ForeignKey(Category, related_name='product', on_delete=models.CASCADE, verbose_name='Категория')
+    description = models.TextField(verbose_name='Описание')
+
+    class Meta:
+        verbose_name = 'Продавец'
+        verbose_name_plural = 'Продавцы'
+
+    def __str__(self):
+        return self.shop_name
+
+
+#The model of basket for products
+class Basket(models.Model):
+    username = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, verbose_name='Пользователь', unique=False)
+    product = models.ForeignKey(Product, on_delete=models.CASCADE, verbose_name='Товары')
+    quantity = models.PositiveIntegerField(default=1, verbose_name='Количество')
```

Comment:
```suggestion
    quantity = models.PositiveIntegerField(default=0, verbose_name='Количество')
```
Why default=1 if initially we don't have products in basket?

Reply:
It was maded as, because when we add a new product to the basket, its quantity should be = 1 (after we can increase or dercrease quantity ), in other case quantity will = 0

Reply:
If quantity of product in basket will be = 0, this product should be deleted, in my opinion